guides/puppet-web-apps/authorization/README.md

```diff
@@ -0,0 +1,227 @@
+```
+
+Rules implement `IAuthorizationRule<TPermission>` interface, which defines one method:
+
+```c#
+public bool Evaluate(
+IEnumerable<Claim> claims, // each Claim represents a fact about the current user. Most popular ones are
+// SystemUserClaim and PersonClaim
+IAuthorizationEnforcement authorizationEnforcement, // this can be used to check if the current user has other permissions
+TPermission permission) // the actual permission to check
+```
+
+`SystemUserClaim` checks weather a claim of specified type (SystemUserClaim) exists and matches the provided predicate. Here, we effectively check if current user has a name "admin".
+
+Claims are a way obtain information about the current user without direct dependency on authentication system. There are two built in claims:
+* SystemUserClaim - if present it means that a current user is signed in as a specific SystemUser
+* PersonClaim - if present it means that a current user represents specific person. It is preferred mean of establishing user's identity, since it's usually more convenient to manage People than Users
+
+There are other built-in rules in `Starcounter.Authorization.Core.Rules` namespace. Feel free to use them or create your own.
+
+### Check the rules
+
+`AuthorizationEnforcement` is a class responsible for checking the if the current user has a specific permission with regard to a rule set. Let's create it using the rules we defined in previous steps:
+
+```c#
+var enforcement = AuthorizationEnforcement(rules, new SystemUserAuthentication());
+```
+
+The second argument passed is a authentication backend (a something that obtains the claims about a current user). It's usually enough to use the built in `SystemUserAuthentication`. It will provide `SystemUserClaim` and `PersonClaim` for singed in users.
+
+We can now use our new `IAuthorizationEnforcement` to check if we have a permission:
+
+```c#
+bool canWe = enforcement.CheckPermission(new ListInvoices()); // note that the outcome depends on weather we are currently signed in as "admin"
+```
+
+## Automatic checking of permissions - SecurityMiddleware
+
+Manual checking of permissions is powerful, but usually we need something that will check the permissions for us. That's what SecurityMiddleware is doing - it's preventing users from accessing pages they don't have access to and from making interactions they are not allowed to do.
+
+### Prerequisite: Router
+
+`SecurityMiddleware` is a feature that depends on Router. If you don't know what it is - go ahead and check it out // `TODO` LINK
+
+To enable `SecurityMiddleware` add it to a Router:
+
+```c#
+router.AddMiddleware(new SecurityMiddleware(
+enforcement,
+info => Response.FromStatusCode(403), // a function returning a Response to give instead of a restricted Page
+PageSecurity.CreateThrowingDeniedHandler<Exception>())); // defines the behavior in case of forbidden interaction
+```
```

Comment:
I would add some real usages of this functions. Like returning page with 404 and creating notification about restricted action.